lib/rubygems/package/tar_reader/entry.rb

```diff
@@ -8,6 +8,21 @@
 # Class for reading entries out of a tar file
 
 class Gem::Package::TarReader::Entry
+  ##
+  # Creates a new tar entry for +header+ that will be read from +io+
+  # If a block is given, the entry is yielded and then closed.
+
+  def self.open(header, io, &block)
+    entry = new header, io
+    return entry unless block_given?
+    begin
+      yield entry
```

Comment:
for consistency with `IO.open`, should the return value of the block be returned from `open` ?

Reply:
Hmm, I think I was imitating File.open here but I'm not certain now. The nil is clearly intentional but I don't remember why for sure. I can check tomorrow.

Reply:
You're completely right. Updated and added a test.